sdk/core/Azure.Core.Experimental/src/Variant/Value.cs

```diff
@@ -0,0 +1,1347 @@
+        /// <summary>
+        /// Casts value to sbyte, if possible.
+        /// </summary>
+        /// <param name="value"></param>
+        public static explicit operator sbyte(in Value value) => value.As<sbyte>();
+        /// <summary>
+        /// Casts nullable sbyte to value.
+        /// </summary>
+        /// <param name="value"></param>
+        public static implicit operator Value(sbyte? value) => new(value);
+        /// <summary>
+        /// Casts value to nullable sbyte, if possible.
+        /// </summary>
+        /// <param name="value"></param>
+        public static explicit operator sbyte?(in Value value) => value.As<sbyte?>();
+        #endregion
+
+        #region Boolean
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="value"></param>
+        public Value(bool value)
+        {
+            this = default;
+            _object = TypeFlags.Boolean;
+            _union.Boolean = value;
+        }
+
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="value"></param>
+        public Value(bool? value)
+        {
+            this = default;
+            if (value.HasValue)
+            {
+                _object = TypeFlags.Boolean;
+                _union.Boolean = value.Value;
+            }
+            else
+            {
+                _object = null;
+            }
+        }
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="value"></param>
+        public static implicit operator Value(bool value) => new(value);
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="value"></param>
+        public static explicit operator bool(in Value value) => value.As<bool>();
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="value"></param>
+        public static implicit operator Value(bool? value) => new(value);
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="value"></param>
+        public static explicit operator bool?(in Value value) => value.As<bool?>();
+        #endregion
+
+        #region Char
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="value"></param>
+        public Value(char value)
+        {
+            this = default;
+            _object = TypeFlags.Char;
+            _union.Char = value;
+        }
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="value"></param>
+        public Value(char? value)
+        {
+            this = default;
+            if (value.HasValue)
+            {
+                _object = TypeFlags.Char;
+                _union.Char = value.Value;
+            }
+            else
+            {
+                _object = null;
+            }
+        }
+        /// <summary>
+        /// TBD.
```

Comment:
Add the real text on these?

Reply:
same for all other instances of "TBD"

Reply:
Yeah, I will do it. But I don't want to invest too much into docs before the API stabilizes just a bit.